### ISSUE TITLE
TabbedShowLayout component resolves path incorreclly if first tab is null

### ISSUE THREAD

**Issue by WiXSL** [marmelab/react-admin#5294]
Trying to set an optional `Tab` component as the first Tab inside a `TabbedShowLayout` component.
**Note**: I solve it by setting a custom path to every Tab and making my own ShowButton.

Test to reproduce the problem in `ra-ui-materialui` package:

The test fails and you get a console error.
`console.error ../../node_modules/@material-ui/core/Tabs/Tabs.js:221
    Material-UI: The value provided to the Tabs component is invalid.
    None of the Tabs' children match with `/`.
    You can provide one of the following values: //1, //2.`

If you remove the `{null}` line, the test pass without errors.

```jsx
import * as React from 'react';
import expect from 'expect';
import { cleanup, render } from '@testing-library/react';

import TabbedShowLayout from './TabbedShowLayout';
import Tab from './Tab';
import TextField from '../field/TextField';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

describe('<TabbedShowLayout />', () => {
    afterEach(cleanup);

    const renderWithRouter = children => {
        const history = createMemoryHistory();

        return {
            history,
            ...render(<Router history={history}>{children}</Router>),
        };
    };

    it('should display the first valid Tab component and its content', () => {
        const { queryByText, history } = renderWithRouter(
            <TabbedShowLayout basePath="/" record={{ id: 123 }} resource="foo">
                {null}
                <Tab label="Tab1">
                    <TextField label="Field On Tab1" source="field1" />
                </Tab>
                <Tab label="Tab2">
                    <TextField label="Field On Tab2" source="field2" />
                </Tab>
            </TabbedShowLayout>
        );

        expect(queryByText('Tab1')).not.toBeNull();
        expect(queryByText('Field On Tab1')).not.toBeNull();
    });
});

**Comment by djhi** [comment]
Have you tried this with raw material-ui Tabs? I think it will behave the same way

**Comment by WiXSL** [comment]
Seems to work.

```jsx
import * as React from 'react';
import expect from 'expect';
import {render, cleanup} from '@testing-library/react';

import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';

const TabPanel = props => {
    const { children, value, index } = props;

    return (
      <div hidden={value !== index}>
          {value === index && <Typography>{children}</Typography>}
      </div>
    );
};

const SimpleTabs = () => {
    const [value, setValue] = React.useState(0);

    const handleChange = (event, newValue) => {
        setValue(newValue);
    };

    return (
      <div>
          <AppBar>
              <Tabs value={value} onChange={handleChange} >
                  {null}
                  <Tab label="Tab1" />
                  <Tab label="Tab2" />
              </Tabs>
          </AppBar>
          {null}
          <TabPanel value={value} index={0}>
              Field On Tab1
          </TabPanel>
          <TabPanel value={value} index={1}>
              Field On Tab2
          </TabPanel>
      </div>
    );
};

describe('<MuiTab />', () => {
    afterEach(cleanup);

   it('should display the first valid Tab component and its content', () => {
        const {queryAllByText} = render(
            <SimpleTabs />
        );

        expect(queryAllByText('Tab1')).toHaveLength(1);
        expect(queryAllByText('Field On Tab1')).toHaveLength(1);
    });
});

**Comment by WiXSL** [comment]
This is a much simplier test

```jsx
import * as React from 'react';
import expect from 'expect';
import { cleanup, render } from '@testing-library/react';

import TabbedShowLayout from './TabbedShowLayout';
import Tab from './Tab';
import TextField from '../field/TextField';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

describe('<TabbedShowLayout />', () => {
    afterEach(cleanup);

    const renderWithRouter = children => {
        const history = createMemoryHistory();

        return {
            history,
            ...render(<Router history={history}>{children}</Router>),
        };
    };

    it('should display the first valid Tab component and its content', () => {
        const { queryByText, history } = renderWithRouter(
            <TabbedShowLayout basePath="/" record={{ id: 123 }} resource="foo">
                {null}
                <Tab label="Tab1">
                    <TextField label="Field On Tab1" source="field1" />
                </Tab>
                <Tab label="Tab2">
                    <TextField label="Field On Tab2" source="field2" />
                </Tab>
            </TabbedShowLayout>
        );

        expect(queryByText('Tab1')).not.toBeNull();
        expect(queryByText('Field On Tab1')).not.toBeNull();
    });
});

**Comment by fzaninotto** [comment]
Fixed by #5312
